### PR TITLE
Allow downloading block stacks as XML

### PR DESCRIPTION
**Pull request by bromagosa**

### CODE REVIEW

**Comment by liam4** [comment]
Next step – copy to and paste from the clipboard.. :)

**Comment by bromagosa** [comment]
That's much harder because we don't have access to the clipboard, but we could probably have an internal `localStorage`-powered blocks clipboard :)

**Comment by cycomachead** [comment]
We do have access to the clipboard. :)

Well, if users Do Cmd-C / Cmd-V, and we could work on a function that calls it, I believe.

File: blocks.js [hunk 1]
```diff
@@ -2593,6 +2593,22 @@ BlockMorph.prototype.userMenu = function () {
         },
         'open a new window\nwith a picture of this script'
     );
+    menu.addLine();
+    menu.addItem(
+        'download stack',
```
Comment (cycomachead) [inline blocks.js]: Do we use the term "stack" anywhere? Why not a "script"? (or am I misunderstanding something?)

**Comment by jmoenig** [comment]
Nah, Michael, the infuriating thing about web programming is that there are all kinds of features available to users that cannot be invoked programmatically. Clipboard is just one of them. :-(

**Comment by cycomachead** [comment]
There has been slowly increasing support for clipboard APIs for a while but fairly recently I think things pretty decent. 

https://w3c.github.io/clipboard-apis/

I use multiple websites where this is possible, and AFAIK they aren't using any of the flash based solutions since I've disabled flash. 

-- 
Michael Ball
From my iPhone
michaelball.co

File: blocks.js (hunk 1 above)
Comment (bromagosa) [inline blocks.js]: Right, "script" would indeed be better :)

Comment (bromagosa) [inline blocks.js]: I changed it to "script" in ee126f1 :)

Comment (jmoenig) [inline blocks.js]: Yay, thanks, Bernat!

**Comment by jmoenig** [comment]
We agreed to pull this one but at first "hide" it so you only see it when you hold the shift key when invoking the context menu. Then we'll play with it and decide when to "unhide" it.
@bromagosa: Can you adjust this PR, and maybe also look into the "receiver" issue in store.js again, because I've changed the structure of scripts to no longer require a receiver.

**Comment by brianharvey** [comment]
[offtopic] Of course we have access to the clipboard, at least for text -- you implemented that so people could paste in their passwords! [/offtopic]

**Comment by jmoenig** [comment]
no, we don't have access to the clipboard. At least not in any programmatical / API way. It's infuriating.

File: gui.js [hunk 2]
```diff
@@ -4086,6 +4089,60 @@ IDE_Morph.prototype.openMediaString = function (str) {
...
+
+    if (Process.prototype.isCatchingErrors) {
+        try {
+            xml = this.serializer.parse(str, this.currentSprite);
+            script = this.serializer.loadScript(xml, this.currentSprite);
+        } catch (err) {
+            this.showMessage('Load failed: ' + err);
+        }
+    } else {
+        xml = this.serializer.loadScript(str, this.currentSprite);
```
Comment (dekrain) [inline gui.js]: Should it be `parse`?

**Pull request closed without merging**